### PR TITLE
Improve image priority and skeleton (fixes #328)

### LINKED ISSUE #328: Improve loading
The page takes a second to load images and content. When it is loaded it looks beautiful, but while loading it's not perfect. The fact the image is missing and there is no loading animation might be the issue.

image

### PR DESCRIPTION
**Pull request by him2312**
The current hero image downloads at a low priority, and shows an empty space. We wanted to reduce image load time and improve the perceived performance by providing a skeleton.

# Pull Request type
- Refactoring (no functional changes, no API changes)

Resolves: #328 

## Changelog
- Add skeleton loader to Feature quest card - [reference](https://github.com/starknet-id/starknet.quest/issues/328#issuecomment-2019252138)
- Set `fetchPriority` to high for hero image, and lazy load images not on first fold.
- Compress webp images to improve the page load time. - [reference](https://github.com/starknet-id/starknet.quest/issues/328#issuecomment-2019254331)
- Hydration error [`<div> cannot appear as a descendent of <p>`](https://stackoverflow.com/questions/41928567/div-cannot-appear-as-a-descendant-of-p)

### CODE REVIEW

**Comment by him2312** [comment]
@ayushtom can you approve the deployment and mark it `Ready to review`?

We can get things reviewed on the preview URL for feedback.

File: components/pages/home/questCategories.tsx [hunk 1]
```diff
@@ -54,10 +54,10 @@ const QuestCategories: FunctionComponent<QuestCategoriesProps> = ({
                   <h2 className="text-gray-200">Boosts Quest</h2>
                   <p className="text-gray-200 normal-case">
                     {completedBoostNumber === boosts.length ? (
-                      <div className="flex">
+                      <span className="flex">
```
Comment (him2312) [inline components/pages/home/questCategories.tsx:57]: There was a hydration issue [`<div> cannot appear as a descendent of <p>`](https://stackoverflow.com/questions/41928567/div-cannot-appear-as-a-descendant-of-p).

This change fixes that.

**Comment by ayushtom** [comment]
Can you specify the improvements here you i can review the changes and verify? Also would help with documenting the improvements brought to app in this PR @him2312

**Comment by him2312** [comment]
@ayushtom Updated the changelog in the PR description.

File: components/UI/featured_banner/featuredQuest.tsx [hunk 2]
```diff
@@ -56,7 +67,9 @@ const FeaturedQuest: FunctionComponent<FeaturedQuestProps> = ({
         </div>
       </div>
       <div className={styles.featuredQuestImageContainer}>
-        <CDNImg src={imgSrc} className={styles.featuredQuestImage} />
+        {imgSrc
+        ? <CDNImage alt={"Feature Quest Image"} src={imgSrc} fill className={styles.featuredQuestImage} priority sizes="(max-width: 1025px) 85vw, 33vw"/>
+        : <Skeleton variant="rectangular" animation="wave" className={styles.featuredQuestImage} sx={{ bgcolor: 'grey.900' }} /> }
```
Comment (ayushtom) [inline components/UI/featured_banner/featuredQuest.tsx]: can you fix the height here? i believe on first load the loading skeleton takes full width but doesn't take up the height of the parent container.

File: components/UI/featured_banner/featuredQuest.tsx [hunk 3]
```diff
@@ -36,12 +37,22 @@ const FeaturedQuest: FunctionComponent<FeaturedQuestProps> = ({
...
       <div className={styles.featuredQuestInfos}>
         <p className={styles.featuredQuestHeading}>{heading}</p>
-        <h3 className={styles.featuredQuestTitle}>{title}</h3>
-        <p className={styles.featuredQuestDescription}>{desc}</p>
+        {title ? <h3 className={styles.featuredQuestTitle}>{title}</h3> : <Skeleton
+            variant="text"
+            width={400}
+            sx={{ fontSize: "2.5rem", bgcolor: "grey.700" }}
+          />}
+        {desc ? <p className={styles.featuredQuestDescription}>{desc}</p> : <Skeleton
```
Comment (ayushtom) [inline components/UI/featured_banner/featuredQuest.tsx]: i would suggest making description height which takes up around 2-3 lines and making title skeleton also take up 2-3 lines. The skeleton takes lesser space than the normally expected title description . We would want to keep these sizes as close to getting the layout right. You can see this behaviour when the `Featured` text shifts from center to top

File: components/pages/home/questCategories.tsx (hunk 1 above)
Comment (ayushtom) [reply components/pages/home/questCategories.tsx:57]: This was nice. Thanks!

File: components/UI/featured_banner/featuredQuest.tsx (hunk 3 above)
Comment (him2312) [reply components/UI/featured_banner/featuredQuest.tsx]: With the title and description being dynamic I wasn't sure of making it a fixed number of lines.

These specifications help. Will make the changes to keep the text shift as less as possible.

**Comment by him2312** [comment]
@ayushtom made the requested changes. Let me know if this looks better.

Desktop:

https://github.com/starknet-id/starknet.quest/assets/11074657/3e7f6357-11d0-4e0f-b75a-01d20bc0a301

Mobile:

https://github.com/starknet-id/starknet.quest/assets/11074657/c3d503e1-8962-4762-b788-bec8d8c2ab12

**Review by ayushtom — APPROVED** [review]
Amazing work on compression and image optimisations which were needed!

**Comment by him2312** [comment]
> Amazing work on compression and image optimisations which were needed!

@ayushtom Thank you. Here's a snapshot of current lighthouse score for before and after comparison.

Screenshot 2024-03-29 at 12 15 42 AM

Also on a side note: I used https://tinypng.com/ to compress the webp images. Something you can try when launching a new quest.

**Pull request merged**